test/e2e/steps/steps.go

```diff
@@ -0,0 +1,676 @@
+package steps
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/cucumber/godog"
+	jsonpatch "github.com/evanphx/json-patch"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-containerregistry/pkg/crane"
+	"github.com/prometheus/common/expfmt"
+	"github.com/prometheus/common/model"
+	"github.com/spf13/pflag"
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/yaml"
+
+	ocv1 "github.com/operator-framework/operator-controller/api/v1"
+)
+
+const (
+	olmNamespace      = "olmv1-system"
```

Comment:
Since the upstream tests are run downstream, this namespace will change, and the catalogd namespace may differ from the operator-controller namespace. At this time, I'm not sure how this is determined, but it's something we need to handle.
Also, some tests are skipped when running the upstream tests downstream, we need to be able to keep that mechanism.